addons/website/static/src/snippets/s_dynamic_snippet/000.js

```diff
@@ -136,9 +136,12 @@ const DynamicSnippet = publicWidget.Widget.extend({
      */
     _prepareContent: function () {
         if (this.$target[0].dataset.numberOfElements && this.$target[0].dataset.numberOfElementsSmallDevices) {
-            this.renderedContent = core.qweb.render(
+            const oldRenderedContent = this.renderedContent;
+            const newRederedContent = core.qweb.render(
                 this.template_key,
                 this._getQWebRenderOptions());
+            this.noRerender = oldRenderedContent.localeCompare(newRederedContent) === 0;
```

Comment:
Seems to have a cost to render a carousel for nothing instead to have something in template like a data-xxx or another way to avoid a useless rendering... no?

Reply:
I totally agree. And I guess we could add an field on the filter to see if we should rerender on call to action or not. Something like `action_rerender` on the `WebsiteSnippetFilter` model ?

This was my first approach as I'm not sure what modifying a model implies when it comes to releasing a fix. I tried looking in for a way to prevent it only from the JavaScript side but I couldn't find a way to easily distinguish between two filters except their ID and I thought that was unreliable. Maybe I'm missing something?

Reply:
your are right, my bad...
It depend of the filter and not the template...
In this case I prefer to use a context in the ir.filter than a new field ;)